### PR TITLE
refactor: apk usage in Dockerfile

### PR DESCRIPTION
**Pull request by PeterDaveHello**
Less commands, faster build, and smaller image!

### CODE REVIEW

**Review by achettyiitr — APPROVED** [review]

**Comment by PeterDaveHello** [comment]
@achettyiitr anything I can do 😅 ?

File: Dockerfile [hunk 1]
```diff
@@ -31,9 +31,8 @@ RUN go build -o rudder-cli ./cmd/rudder-cli/
 
 FROM alpine:${ALPINE_VERSION}
 
-RUN apk update && apk add tzdata
-RUN apk -U --no-cache upgrade && \
-    apk add --no-cache ca-certificates postgresql-client curl bash
```
Comment (achettyiitr) [inline Dockerfile:36]: Any reason, we are skipping update here?

**Review by achettyiitr — COMMENTED** [review]
Any reason, we are skipping the update here?

File: Dockerfile [hunk 2]
```diff
@@ -31,9 +31,8 @@ RUN go build -o rudder-cli ./cmd/rudder-cli/
...
+RUN apk --no-cache upgrade && \
```
Comment (atzoum) [inline Dockerfile:34]: It is a good idea to always do an update right before doing an upgrade or add command. That way the command will install the latest available packages. [link](https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper#:~:text=Tip%3A%20With%20remote%20repositories%2C%20it%20is%20a%20good%20idea%20to%20always%20do%20an%20update%20right%20before%20doing%20an%20upgrade%20or%20add%20command.%20That%20way%20the%20command%20will%20install%20the%20latest%20available%20packages.)
```suggestion
RUN apk --no-cache update && apk --no-cache upgrade && \
```

Comment (PeterDaveHello) [reply Dockerfile:34]: Not exactly, `--no-cache` will handle that for you, and leave no temporary files.

**Comment by PeterDaveHello** [comment]
@achettyiitr @atzoum just FYR, it's kind of Dockerfile best practices:

https://github.com/hadolint/hadolint/wiki/DL3019

**Comment by PeterDaveHello** [comment]
Not sure if there's anything I can do here?

File: Dockerfile (hunk 2 above)
Comment (atzoum) [reply Dockerfile:34]: the request was to run `apk --no-cache update` before `apk --no-cache upgrade`, not sure how your response is relevant

Comment (PeterDaveHello) [reply Dockerfile:34]: `--no-cache` actually will make `apk` ignore any local cache and fetch the latest index without local temp files needed to be cleaned up

**Review by atzoum — APPROVED** [review]

**Comment by atzoum** [comment]
Thank you @PeterDaveHello, have you performed this step?
https://github.com/rudderlabs/rudder-server/blob/master/CONTRIBUTING.md#rudderstack-contributor-agreement

**Comment by PeterDaveHello** [comment]
It's done now!

**Comment by lvrach** [comment]
Thanks @PeterDaveHello for the contribution

**Pull request merged**